File: src/Domain/Tournaments/Commands.fs
namespace Orleans.Tournament.Domain.Tournaments
open System
open Orleans.Tournament.Domain.Abstractions

type CreateTournament =
    { Name:string
      TournamentId:Guid
      TraceId:Guid
      InvokerUserId:Guid }
    interface ITraceable with
        member x.TraceId = x.TraceId
        member x.InvokerUserId = x.InvokerUserId

type AddTeam =
    { TournamentId:Guid
      TeamId:Guid
      TraceId:Guid
      InvokerUserId:Guid }
        interface ITraceable with
          member x.TraceId = x.TraceId
          member x.InvokerUserId = x.InvokerUserId

type StartTournament =
    { TournamentId:Guid
      TraceId:Guid
      InvokerUserId:Guid }
        interface ITraceable with
          member x.TraceId = x.TraceId
          member x.InvokerUserId = x.InvokerUserId

type SetMatchResult =
    { TournamentId:Guid
      MatchInfo:MatchInfo
      TraceId:Guid
      InvokerUserId:Guid }
        interface ITraceable with
          member x.TraceId = x.TraceId
          member x.InvokerUserId = x.InvokerUserId

File: src/DomainC/Tournaments/Commands.cs
using Tournament.Domain.Abstractions;

namespace Tournament.Domain.Tournaments;

public record CreateTournament(
    string Name,
    Guid TournamentId,
    Guid TraceId,
    Guid InvokerUserId) : ITraceable;

public record AddTeam(
    Guid TournamentId,
    Guid TeamId,
    Guid TraceId,
    Guid InvokerUserId) : ITraceable;

public record StartTournament(
    Guid TournamentId,
    Guid TraceId,
    Guid InvokerUserId) : ITraceable;

public record SetMatchResult(
    Guid TournamentId,
    Match Match,
    Guid TraceId,
    Guid InvokerUserId) : ITraceable;